File: 104. Maximum Depth of Binary Tree/solution_bfs.py
from typing import *
from collections import deque

class TreeNode:
    """Definition for a binary tree node."""
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        q = deque()

        if root:
            q.append(root)

        depth = 0
        while q:
            depth += 1
            current_length = len(q)
            
            for _ in range(current_length):
                node = q.popleft()

                node.left and q.append(node.left)
                node.right and q.append(node.right)
        
        return depth

File: 104. Maximum Depth of Binary Tree/solution_bfs.js
/** Definition for a binary tree node. */
function TreeNode(val, left, right) {
    this.val = (val === undefined ? 0 : val)
    this.left = (left === undefined ? null : left)
    this.right = (right === undefined ? null : right)
}

/**
 * @param {TreeNode} root
 * @return {number}
 */
var maxDepth = function (root) {
    const queue = [];
    if (root) {
        queue.push(root);
    }
    
    let levels = 0;

    while (queue.length > 0) {
        levels++;
        const currentLength = queue.length;

        for (let i = 0; i < currentLength; i++) {
            const node = queue.shift();

            node.left && queue.push(node.left);
            node.right && queue.push(node.right);
        }
    }

    return levels;
};